package symphony.proto;
message DBRoleData {
    optional uint64      AccountID    = 1;
    optional uint64      RoleID       = 2;
    optional int32       CarrerID     = 3;
    optional int32       Level        = 4;
    optional int64       Exp          = 5;
    optional string      Name         = 6;
    optional int32       LangID       = 7;
    optional int32       VipLvl        = 8;
    optional int32       VipExp       = 9;
    optional int64       FightValue   = 10;
    optional int32       CityCopyID   = 11;
    repeated int64 Action = 12;
    repeated int64 Actime = 13;
    optional uint64      CreateTime   = 14;
    optional uint64      LogonTime    = 15;
    optional uint64      LogoffTime   = 16;
    optional uint64      GuildID      = 17;
    optional int32       Channel      = 18;
    optional uint32      OnlineTime    = 19;
}

message DBBagItem {
    optional uint64    Guid = 1;            //格子的guid
    optional uint64    RoleID = 2;            //角色ID
    optional uint64    ItemGuid = 3;        //物品guid
    optional uint32    ItemID = 4;            //道具ID
    optional int64     Count = 5;            //个数
    optional int32     Bind = 6;            //是否是邦定
    optional int32     Status = 7;            //是否删除
}

message DBBagData {
    repeated DBBagItem ItemList = 1;
}

message DBCopyItem {
    optional uint64  RoleID = 1;            //角色ID
    optional uint32  CopyID = 2;
    optional int32   StarNum = 3;
    optional int32   BattleCnt = 4;
    optional int32   ResetCnt = 5;
    optional uint64  BattleTime = 6;
    optional uint64  ResetTime = 7;
}

message DBChapterItem {
    optional uint64  RoleID = 1;            //角色ID
    optional uint32  CopyType = 2;
    optional uint32  ChapterID = 3;
    optional uint32  StarAward = 4;
    optional uint32  SceneAward = 5;
}

message DBCopyData {
    repeated DBCopyItem CopyList = 1;
    repeated DBChapterItem ChapterList = 2;
}

message DBEquipItem {
    optional uint64   RoleID = 1;         //角色ID
    optional uint64   Guid = 2;
    optional int32    EquipID = 3;
    optional int32    StrengthLvl = 4;   //强化等级
    optional int32    RefineLevel = 5;   //精炼等级
    optional int32    StarLevel = 6;      //星级
    optional int32    RefineExp = 7;      //精验的经验
    optional int32    StarExp = 8;      //星级经验
    optional bool     IsUsing = 9;       //是否使用中
}

message DBEquipData {
    repeated    DBEquipItem EquipList = 1;
}

message DBGemItem {
    optional uint64   RoleID = 1;         //角色ID
    optional uint64   Guid = 2;
    optional int32    GemID = 3;
    optional int32    StrengthLvl = 4;   //强化等级
    optional int32    RefineLevel = 5;   //精炼等级
    optional int32    StarLevel = 6;      //星级
    optional int32    RefineExp = 7;      //精验的经验
    optional int32    StarExp = 8;          //星级经验
    optional int32    Pos = 9;        //位置
}

message DBGemData {
    repeated    DBGemItem GemList = 1;
}


message DBSkillItem {
    optional uint64   RoleID = 1;         //角色ID
    optional uint32   SkillID = 2;        //技能ID
    optional int32    Level = 3;          //技能等级
    optional int32    KeyPos = 4;         //按键位置
}

message DBSkillData {
    repeated    DBSkillItem SkillList = 1;
}


message DBPetItem {
    optional uint64   RoleID = 1;         //角色ID
    optional uint64   Guid = 2;
    optional int32    PetID = 3;
    optional int32    StrengthLvl = 4;   //强化等级
    optional int32    RefineLevel = 5;   //精炼等级
    optional int32    StarLevel = 6;      //星级
    optional bool     IsUsing = 7;       //是否使用中
}

message DBPetData {
    repeated    DBPetItem PetList = 1;
}

message DBMountItem {
    optional uint64   RoleID = 1;         //角色ID
    optional uint64   Guid = 2;
    optional int32    MountID = 3;
    optional int32    StrengthLvl = 4;   //强化等级
    optional int32    RefineLevel = 5;   //精炼等级
    optional int32    StarLevel = 6;      //星级
    optional bool     IsUsing = 7;       //是否使用中
}

message DBMountData {
    repeated    DBMountItem MountList = 1;
}

message DBPartnerItem {
    optional uint64   RoleID = 1;         //角色ID
    optional uint64   Guid = 2;
    optional int32    PartnerID = 3;
    optional int32    StrengthLvl = 4;   //强化等级
    optional int32    RefineLevel = 5;   //精炼等级
    optional int32    StarLevel = 6;      //星级
    optional int32    SetPos = 7;       //上阵位置
}

message DBPartnerData {
    repeated DBPartnerItem PartnerList = 1;
}

message DBTaskItem {
    optional uint64  RoleID = 1;            //角色ID
    optional uint32  TaskID = 2;
    optional int32   Status = 3;
    optional int32   Progress = 4;
}

message DBTaskData {
    repeated DBTaskItem TaskList = 1;
}

message DBMailItem {
    optional uint64   RoleID    = 1;   //角色ID
    optional uint64   Guid      = 2;   //邮件guid
    optional uint64   GroupID   = 3;   //群邮件guid
    optional uint64   Time      = 4;   //邮件时间
    optional uint64   SenderID  = 5;   //发送者ID
    optional string   Sender    = 6;   //发送者名字
    optional uint32   MailType  = 7;
    optional uint32   Status    = 8;
    optional string   Title     = 9;   //邮件标题
    optional string   Content   = 10;  //邮件内容
    optional bytes    Items     = 11;  //邮件物品
}

message DBMailData {
    repeated DBMailItem MailList = 1;
}

message DBActivityItem {
    optional uint64    RoleID = 1;            //角色ID
    optional uint32    ActivityID = 2;  //活动ID
    optional uint32    ActivityType = 3;//活动类型
    optional uint32    JoinTime = 4;    //参与时间
    optional uint32    DataLen = 5;     //数据时间长度
    optional bytes     Data = 6;    //活动数据
}

message DBActivityData {
    repeated DBActivityItem ActivityList = 1;
}


message DBCounterItem {
    optional uint64    RoleID       = 1;        //角色ID
    optional uint32    CounterID    = 2;        //计数器ID
    optional uint32    Index        = 3;        //子索引
    optional int64     Value        = 4;        //计数器值
    optional uint64    Time         = 5;        //参与时间
    optional uint32    Param1       = 6;        //参数1
    optional uint32    Param2       = 7;        //参数2
}

message DBCounterData {
    repeated DBCounterItem CounterList = 1;
}


message DBFriendItem {
    optional uint64    RoleID        = 1;        //角色ID
    optional uint64    FriendID      = 2;        //好友ID
}

message DBFriendData {
    repeated DBFriendItem FriendList = 1;
}

message DBRoleLoginAck {
    optional uint32              RetCode     = 1;
    optional uint64              RoleID      = 2;
    optional DBRoleData          RoleData    = 3;
    optional DBBagData           BagData     = 4;
    optional DBCopyData          CopyData    = 5;
    optional DBEquipData         EquipData   = 6;
    optional DBGemData           GemData     = 7;
    optional DBPetData           PetData     = 8;
    optional DBPartnerData       PartnerData = 9;
    optional DBTaskData          TaskData    = 10;
    optional DBMailData          MailData    = 11;
    optional DBMountData         MountData   = 12;
    optional DBActivityData      ActivityData= 13;
    optional DBCounterData       CounterData = 14;
    optional DBFriendData        FriendData  = 15;
    optional DBSkillData         SkillData   = 16;
}